apps/server/src/modules/user/service/user.service.ts

```diff
@@ -39,6 +42,42 @@ export class UserService {
 		return userDto;
 	}
 
+	async getResolvedUser(userId: EntityId): Promise<ICurrentUser> {
+		const user: User = await this.userRepo.findById(userId, true);
+		const account: AccountDto = await this.accountService.findByUserIdOrFail(userId);
+
+		const permissions = new Set<string>();
+		if (user.roles) {
+			user.roles.getItems().forEach((role) => {
+				if (role.permissions) {
+					role.permissions.forEach((permission) => {
+						permissions.add(permission.toString());
+					});
+				}
+			});
+		}
+
+		return {
+			accountId: account.id,
+			systemId: account.systemId,
+			roles: user.roles.getItems().map((role: Role): EntityId => role.id),
+			schoolId: user.school.id,
+			userId: user.id,
+			user: {
+				id: user.id,
+				createdAt: user.createdAt,
+				updatedAt: user.updatedAt,
+				firstName: user.firstName,
+				lastName: user.lastName,
+				roles: user.roles.getItems().map((role: Role): IRole => {
+					return { id: role.id, name: role.name };
+				}),
+				schoolId: user.school.id,
+				permissions: Array.from(permissions),
+			},
+		};
```

Comment:
Is there a way to get this into a seperate function?

Reply:
like a mapper function?